src/encoded/static/components/matrix_chip_seq.js

```diff
@@ -0,0 +1,906 @@
+import { SearchFilter } from './matrix';
+import { TextFilter } from './search';
+import DataTable from './datatable';
+
+
+const SEARCH_PERFORMED_PUBSUB = 'searchPerformed';
+const SEARCH_RESULT_COUNT_PUBSUB = 'searchResultCount';
+const CLEAR_SEARCH_BOX_PUBSUB = 'clearSearchBox';
+const LARGE_MATRIX_MIN_SIZE = 150;
+
+/**
+ * Tranform context to a form where easier to fetch information
+ *
+ * @param {context} context - Context from react
+ * @param {string} assayTitle - Assay Title
+ * @param {string} organismName - Organism Name
+ * @returns {object} - Object with structure - { chIPSeqData, subTabs };
+ *
+ *      chIPSeqData is an object where:
+ *          key is a sub Tab
+ *          value is of structure - { headerRow, dataRow, assayTitle, organismName };
+ *              headerRow - header content-list
+ *              dataRow - array of non-header content
+ *                  key: biosample ontology classification
+ *                  value: array containing counts
+ *              assayTitle - Assay title
+ *              organismName- Organism name
+ *      subTabs: List of subTabs for easy access
+ */
+const getChIPSeqData = (context, assayTitle, organismName) => {
+    if (!context || !context.matrix || !context.matrix.x || !context.matrix.y || !assayTitle || !organismName) {
+        return null;
+    }
+
+    const subTabSource = 'biosample_ontology.classification';
+    const subTabs = context.matrix.x[subTabSource].buckets.map(x => x.key);
+    const chIPSeqData = {};
+
+    subTabs.forEach((subTab) => {
+        const xGroupBy1 = context.matrix.x.group_by[0];
+        const xGroupBy2 = context.matrix.x.group_by[1];
+        const headerRow = context.matrix.x[xGroupBy1].buckets.find(f => f.key === subTab)[xGroupBy2]
+            .buckets
+            .reduce((a, b) => {
+                const m = a.concat(b);
+                return m;
+            }, [])
+            .map(x => x.key);
+        const headerRowIndex = headerRow.reduce((x, y, z) => { x[y] = z; return x; }, []);
+        const headerRowLength = headerRow.length;
+        const yGroupBy1 = context.matrix.y.group_by[0];
+        const yGroupBy2 = context.matrix.y.group_by[1];
+
+        const yData = context.matrix.y[yGroupBy1].buckets
+            .find(rBucket => rBucket.key === organismName)[yGroupBy2].buckets
+            .reduce((a, b) => {
+                const m = {};
+                m[b.key] = b[xGroupBy1].buckets
+                    .filter(f => f.key === subTab)
+                    .reduce((x, y) => {
+                        x.push(y[xGroupBy2].buckets
+                            .reduce((i, j) => i.concat(j), []));
+                        return x;
+                    }, []);
+                return a.concat(m);
+            }, []);
+
+        const dataRowT = {};
+
+        yData.forEach((y) => {
+            const yKey = Object.keys(y)[0];
+            dataRowT[yKey] = dataRowT[yKey] || [...Array(headerRowLength + 1)].fill(0);
```

Comment:
Left as-is. The [...] is needed for the array to be of size headerRowLength + 1, otherwise, it is a single value. Let me know if you still have concerns and we can go from there.

Reply:
I think it’s the same without the brackets.
<img width="311" alt="image" src="https://user-images.githubusercontent.com/1181324/74469626-e2a4f380-4e51-11ea-9e9b-bda9a5eccdae.png">

Reply:
Done. I was doing [Array(5)] rather than Array(5).